cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/runtime/schedule/trigger/TriggeringInfoHelper.java

```diff
@@ -0,0 +1,181 @@
+package io.cdap.cdap.internal.app.runtime.schedule.trigger;
+
+import com.google.gson.Gson;
+import com.google.gson.JsonDeserializationContext;
+import com.google.gson.JsonDeserializer;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParseException;
+import com.google.gson.JsonSerializationContext;
+import com.google.gson.JsonSerializer;
+import io.cdap.cdap.api.schedule.Trigger;
+import io.cdap.cdap.api.schedule.TriggerInfo;
+import io.cdap.cdap.internal.app.runtime.schedule.DefaultTriggeringScheduleInfo;
+import io.cdap.cdap.proto.ProgramType;
+import io.cdap.cdap.proto.TriggeringInfo;
+import io.cdap.cdap.proto.TriggeringPropertyMapping;
+import io.cdap.cdap.proto.id.ProgramRunId;
+import io.cdap.cdap.proto.id.ScheduleId;
+
+import java.lang.reflect.Type;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+/**
+ * Helper functions for {@link TriggeringInfo}
+ */
+public class TriggeringInfoHelper {
+
+  private static Gson gson = new Gson();
+  private static final String TRIGGERING_PROPERTIES_MAPPING = "triggering.properties.mapping";
+
+  /**
+   * Create a new {@link TriggeringInfo} object given
+   * {@link DefaultTriggeringScheduleInfo}, {@link ScheduleId} and {@link Trigger.Type}
+   * @param triggeringScheduleInfo
+   * @param type
+   * @param scheduleId
+   * @return
+   */
+  public static TriggeringInfo fromTriggeringScheduleInfo(DefaultTriggeringScheduleInfo triggeringScheduleInfo,
+                                                          Trigger.Type type, ScheduleId scheduleId) {
+    List<TriggerInfo> triggerInfos = triggeringScheduleInfo.getTriggerInfos();
+    switch(type) {
+      case PROGRAM_STATUS:
+        DefaultProgramStatusTriggerInfo programStatusTriggerInfo =
+          (DefaultProgramStatusTriggerInfo) triggerInfos.get(0);
```

Comment:
Is it expected to be only a single element since it is non-composite? Please add a comment.